lua/profile/util.lua

```diff
@@ -5,6 +5,46 @@ local tbl_isarray = vim.isarray or vim.tbl_isarray or vim.tbl_islist
 local pack_len = vim.F.pack_len
 local split = vim.split
 
+
+M.DEFAULT_PROCESS_ID = 1
+M.DEFAULT_THREAD_ID = 1
+
+if jit and jit.version then
+  local ffi = require("ffi")
+
+  if vim.fn.has("win32") == 1 or vim.fn.has("win64") == 1 then
+    ---@return number # The current thread's ID number (1-or-more value).
+    function M.get_thread_id()
+      ffi.cdef([[
+        typedef unsigned long DWORD;
+        DWORD GetCurrentThreadId();
+      ]])
+
+      return ffi.C.GetCurrentThreadId()
+    end
+  else
+    ---@return number # The current thread's ID number (1-or-more value).
+    function M.get_thread_id()
+      ffi.cdef([[
+        typedef int pid_t;
+        pid_t gettid();
+      ]])
```

Comment:
Could we move the cdefs outside of the function call? Or make it a once-only call? Running the C parser for every single event will bog us down a lot and skew the results.

Reply:
You're right my bad. I'll make it once-only at best and if not make it global so it isn't calling each time the function runs

Reply:
I thought about it and if you don't mind let's do a global definition instead. A run-once makes sense in a situation where there's a good chance the function isn't called but since it's part of the main profiler it's going to be called pretty much for sure. We might as well skip the run-once overhead.

Reply:
Yep, I agree this is simpler and doesn't have much downside